timeseries/blueflood/blueflood.go

```diff
@@ -474,101 +385,56 @@ var samplerMap = map[timeseries.SampleMethod]sampler{
 	},
 }
 
-// ChooseResolution will use the finest-grained resolution which doesn't exceed the slot limit.
-// Thus, if you request too many points, it will automatically reduce the resolution.
-func (b *Blueflood) ChooseResolution(requested api.Timerange, smallestResolution time.Duration) time.Duration {
-	// In some cases, coarser-resolution data may have a shorter TTL.
-	// To accommodate these cases, it must be verified that the requested timerange will
-	// actually be present for the chosen resolution.
-	// TODO: figure out how to make this work with moving averages and timeshifts
-
-	requiredAge := b.config.TimeSource().Sub(requested.Start())
-
-	for _, resolution := range Resolutions {
-		survivesFor := b.config.oldestViableDataForResolution(resolution)
-		if survivesFor < requiredAge {
-			// The data probably won't be around for the earliest part of the timerange,
-			// so don't use this resolution
-			continue
-		}
-		if resolution.duration < requested.Resolution() {
-			// Skip this timerange, it is finer than the one requested.
-			continue
-		}
-		// Check that the timerange is large enough
-		if resolution.duration >= smallestResolution {
-			return resolution.duration
-		}
+func (b *Blueflood) FetchMultipleTimeseries(request timeseries.FetchMultipleRequest) (api.SeriesList, error) {
+	defer request.Profiler.Record("Blueflood FetchMultipleTimeseries")()
+
+	singleRequests := request.ToSingle()
+	results := make([]api.Timeseries, len(singleRequests))
+	errs := make([]error, len(singleRequests))
```

Comment:
You're right- I forgot to get rid of it when I switched to using the queue.

Deleted.